rtsp.go

```diff
@@ -635,6 +641,19 @@ func newRTSPCamera(ctx context.Context, _ resource.Dependencies, conf resource.C
 		logger.Error(err.Error())
 		return nil, err
 	}
+
+	// initialize a pool to manage AVFrames so they don't bog down the garbage collector
+	framePool := &sync.Pool{
+		New: func() interface{} {
+			avFrame, err := allocateAVFrame()
+			if err != nil {
+				logger.Errorf("Failed to allocate AVFrame in frame pool: %v", err)
+				return nil
```

Comment:
what happens if `nil` is returned from this function?

Reply:
```
	d.dst = d.avFramePool.Get().(*avFrameWrapper)
	if d.dst == nil {
		return nil, errors.New("failed to obtain AVFrame from pool")
	}
```

We should catch it here. `.Get()` calls `.New()` when there are no existing, available items in the pool to return